### ISSUE TITLE
pshtt incorrectly labeling www as the canonical_endpoint when root redirects to *any* www URL

### ISSUE THREAD

**Issue by PaulSD** [dhs-ncats/pshtt#152]
search.cio.gov is used to present search results for cio.gov.  If a valid search query is not provided, search.cio.gov redirects the user back to https://www.cio.gov/.
A *.cio.gov wildcard DNS record and certificate are used for search.cio.gov.  Due to the differences in wildcard handling between DNS and certificates, *.search.cio.gov subdomains will resolve, but HTTPS requests to them will result in a certificate error.
If we were expecting users to type this URL manually, we would likely add www.search.cio.gov as a SAN on the certificate to prevent that commonly-used prefix from resulting in a certificate error.  However, in this case, users are never expected to enter this URL directly, so we have no need to gracefully handle such quirks in user behavior.

When pshtt checks this domain, canonical_endpoint() detects the redirect being returned by search.cio.gov and tries to perform validation on www.search.cio.gov instead, which of course fails.  This causes search.cio.gov to incorrectly show up as non-compliant in Pulse.

**Comment by konklone** [comment]
Your analysis looks right to me. Here's the latest pshtt results for `search.cio.gov`:

* https://s3-us-gov-west-1.amazonaws.com/cg-4adefb86-dadb-4ecf-be3e-f1c7b4f6d084/live/subdomains/scan/cache/pshtt/search.cio.gov.json

I think the issue is that we have a naive check for whether or not the root endpoint is seen as "immediately redirecting to the www endpoint", during [our redirect analysis](https://github.com/dhs-ncats/pshtt/blob/develop/pshtt/pshtt.py#L337-L345):

```python
endpoint.redirect_immediately_to_www = (re.match(r'^https?://www\.', immediate) is not None)
```

This is _meant_ to catch redirects specifically to the `www` version of the host being scanned (in this case, `www.search.cio.gov`), but it's catching the redirect to `www.cio.gov` as well. That should be fixable.

**Comment by konklone** [comment]
This is fixed by #154.
